filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -65,23 +66,25 @@ class BlobFileSystemManager(val container: BlobContainerName,
   private var expiry: Option[Instant] = initialExpiration
 
   def getExpiry: Option[Instant] = expiry
-  def uri: URI = BlobFileSystemManager.uri(endpoint)
   def isTokenExpired: Boolean = expiry.exists(BlobFileSystemManager.hasTokenExpired(_, buffer))
-  def shouldReopenFilesystem: Boolean = isTokenExpired || expiry.isEmpty
-  def retrieveFilesystem(): Try[FileSystem] = {
+  def shouldReopenFilesystem(container: BlobContainerName): Boolean = isTokenExpired || expiry.isEmpty || !lastOpenContainer.contains(container.value)
```

Comment:
Should we be checking the endpoint as well as the container? In theory you could have two containers with the same name in different storage accounts.